Migrate some icons from Font Awesome icon font to SVG version

liamdawson opened the pull request:
**2018-09-24:**
Opening this PR to get an idea if I'm going an agreeable direction to start with.

**2018-09-29:**
Have Angular and React components, ready to migrate icons in earnest.

**2018-10-06:**
Removing WIP tag - big enough and ugly enough to merge to get some of them over the line.

liamdawson commented:
(This is for #2089)

bhollis commented:
Thanks for tackling this!

I think the tough part is going to be all the usages inside the AngularJS code. Do you have any ideas how to handle those?

liamdawson commented:
Honestly, I don't have an answer for that right now. I was planning on investigating the core lib underneath (the FortAwesome SVG base) to see if I could use it to emit the correct output, but it's still going to be a bit messy at best.

SunburnedGoose commented:
What is the difference between ActionIcon and StaticIcon aside from the additional class name for ActionIcon?

liamdawson commented:
No difference. At the moment, it's record keeping for me in terms of desired behaviour - `StaticIcon` elements (semantically) don't change, where `ActionIcon` elements will have an on-hover effect.

I'm not attached to it, so I could easily end up collapsing back to one element by the time I finish.

liamdawson commented:
Updated with an Angular directive, and simplified structure and usage of icons. Even though the new stuff is SVG, it's really good about respecting CSS colours/sizing etc, so there's no need for half the props I was passing.

SunburnedGoose commented:
This looks like it's in a great place.  You're importing all the items you need in one place, with clear React and AngularJS components.  Fly!

bhollis commented on src/app/shell/icons/AppIcon.tsx:
```diff
@@ -0,0 +1,15 @@
+  className?: string;
+}
+
+export const AppIconComponent: React.SFC<IComponentProps> = ({ icon, className }) => (
```
I know it's weird, but in the rest of the codebase we would write this as:

```ts
export default function AppIconComponent({
  icon, 
  className
}: {
  icon: any;
  className?: string;
}) {
  return <FontAwesomeIcon className={'app-icon ' + (className || '')} icon={icon} />;
}
```

liamdawson commented:
@bhollis I'm ready for the full/final review on this PR, it does a good chunk of the React components and some of the angular ones. Next one will do the remaining React, then Angular after that.

bhollis commented on src/app/bungie-api/error-toaster.ts:
```diff
@@ -1,13 +1,17 @@
+  const twitterIcon = `<span style="font-size: 1.8em; vertical-align: middle;">${renderToHtmlByLibraryName(
+    'twitterIcon'
+  )}</span>`;
+  console.log(twitterIcon);
```
Leftover console.log
